src/extension/provider/statefulAuth.ts

```diff
@@ -248,6 +237,90 @@ export class StatefulAuthProvider implements AuthenticationProvider, Disposable
     this.#disposables.forEach((d) => d.dispose())
   }
 
+  public async bootstrapFromToken() {
+    const authTokenUri = Uri.joinPath(this.context.extensionUri, 'secrets', 'authToken')
+
+    if (!this.hasAuthTokenFile(authTokenUri)) {
+      logger.info('No auth token file found, halting bootstrap from token.')
+      return
+    }
+
+    try {
+      const { token, payload } = await this.insecureDecode(authTokenUri)
+      const session = await this.buildSession(token, payload)
+      await this.persistSessions([session], { added: [session], removed: [], changed: [] })
+      await this.deleteAuthTokenFile(authTokenUri)
+      return true
```

Comment:
Return true here but `undefined` everywhere else. I suggest adding a `: boolean` to the return type of the function and making sure it's consistent.

Semantically, undefined and false are the same but it's sloppy typing.